projects/packages/waf/src/class-waf-stats.php

```diff
@@ -69,4 +76,86 @@ public static function get_rules_version() {
 	public static function get_automatic_rules_last_updated() {
 		return get_option( Waf_Rules_Manager::AUTOMATIC_RULES_LAST_UPDATED_OPTION_NAME );
 	}
+
+	/**
+	 * Checks if the current cached global stats is expired and should be renewed
+	 *
+	 * @return boolean
+	 */
+	public static function is_global_stats_cache_expired() {
+		$option_timestamp = get_option( 'jetpack_protect_global_stats_timestamp' );
+
+		if ( ! $option_timestamp ) {
+			return true;
+		}
+
+		return time() > (int) $option_timestamp;
+	}
+
+	/**
+	 * Checks if we should consider the stored cache or bypass it
+	 *
+	 * @return boolean
+	 */
+	public static function should_use_global_stats_cache() {
+		return ! ( defined( 'JETPACK_PROTECT_DEV__BYPASS_CACHE' ) && JETPACK_PROTECT_DEV__BYPASS_CACHE );
+	}
+
+	/**
+	 * Get the global stats from the API endpoint
+	 */
+	public static function fetch_global_stats_from_api() {
+		// todo: merge and deploy add/wpcom-endpoint-for-wpscan-statistics first!
+		$url      = esc_url_raw( 'https://public-api.wordpress.com/wpcom/v2/jetpack-protect-global-stats' );
+		$response = wp_remote_get( $url );
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+
+		if ( is_wp_error( $response ) || 200 !== $response_code || empty( $response['body'] ) ) {
+			return new \WP_Error( 'failed_fetching_global_stats', 'Failed to fetch global stats data from the server', array( 'status' => $response_code ) );
+		}
+
+		$body = json_decode( wp_remote_retrieve_body( $response ) );
+
+		update_option( 'jetpack_protect_global_stats', maybe_serialize( $body ) );
+		update_option( 'jetpack_protect_global_stats_timestamp', time() + 86400 ); // Caches expires after 24 hours.
+
+		return $body;
+	}
+
+	/**
+	 * Gets the current cached global stats
+	 *
+	 * @return bool|array False if value is not found. Array with values if cache is found.
+	 */
+	public static function get_global_stats_from_options() {
+		return maybe_unserialize( get_option( 'jetpack_protect_global_stats' ) );
+	}
+
+	/**
+	 * Get the global stats
+	 * If the cache is expired, it will fetch the data from the API
+	 * If the cache is not expired, it will return the cached data
+	 *
+	 * @param bool $refresh_from_wpcom Whether to refresh the data from the API.
+	 * @return array|\WP_Error
+	 */
+	public static function get_global_stats( $refresh_from_wpcom = false ) {
+		if ( self::$global_stats !== null ) {
+			return self::$global_stats;
+		}
+
+		if ( $refresh_from_wpcom || ! self::should_use_global_stats_cache() || self::is_global_stats_cache_expired() ) {
+			$global_stats = self::fetch_global_stats_from_api();
+		} else {
+			$global_stats = self::get_global_stats_from_options();
+		}
+
+		// Ensure that $global_stats is not null and is of the correct type
+		if ( $global_stats === null || ( ! is_array( $global_stats ) && ! ( $global_stats instanceof \WP_Error ) ) ) {
```

Comment:
```suggestion
		// Ensure that $global_stats is the correct type
		if ( ! is_array( $global_stats ) && ! ( $global_stats instanceof \WP_Error ) ) {
```
My thinking here - if the stats variable must either be an array or a WP_Error, we don't need to check for null as that is covered by it not being an array or instance of WP_Error.

Reply:
I added this to resolve a phan error originally, I've applied the suggested changes, but if that error is raised again I may need to suppress it since we can certain we have all angles covered.